storage/gcsemu/gcsemu.go

```diff
@@ -122,6 +126,18 @@ func (g *GcsEmu) Handler(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	if r.Header.Get("content-encoding") == "gzip" {
+		gzipReader, err := gzip.NewReader(r.Body)
+		if err != nil && io.EOF != err {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		if err == nil {
+			r.Body = gzipReader
+		}
+	}
+
```

Comment:
This seems like a pretty solid change in isolation.  I assume this is "Fix gzip handling in java sdk problem in requests including creating bucket"?